### PR TITLE
[sailfish-components-webview] Implement new dialog for Geolocation API permission request. Fixes JB#51603

### PR DESCRIPTION
**Pull request by paveltkv**

### CODE REVIEW

File: import/pickers/PickerOpener.qml [hunk 1]
```diff
@@ -74,7 +74,9 @@ QtObject {
 
     Component.onCompleted: {
         if (contentItem) {
-            contentItem.addMessageListeners(listeners)
+            for (var i = 0; i < listeners.length; ++i) {
+                contentItem.addMessageListener(listeners[i])
+            }
```
Comment (atatarov) [inline import/pickers/PickerOpener.qml:79]: may be it would be better
```
for (iter in listeners) {
  contentItem.addMessageListener(listeners[iter])
}
```

File: import/popups/LocationDialog.qml [hunk 2]
```diff
@@ -17,17 +18,44 @@ UserPrompt {
...
-        //% "The site '%1' wants to use your current location."
-        text: qsTrId("sailfish_components_webview_popups-la-location_requested").arg(host)
-        wrapMode: Text.WordWrap
-        color: Theme.highlightColor
+    //: Deny the server to use location
+    //% "Deny"
+    cancelText: qsTrId("sailfish_components_webview_popups-he-deny_location")
+
+    Column {
+        id: contentColumn
```
Comment (atatarov) [inline import/popups/LocationDialog.qml:32]: it seems this id is not used anywhere

**Comment by rainemak** [comment]
Re-target to master and rebase

File: import/popups/LocationDialog.qml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2013 Jolla Ltd.
+** Copyright (c) 2013 Jolla Ltd.
```
Comment (rainemak) [inline import/popups/LocationDialog.qml:3]:
```suggestion
** Copyright (c) 2013 - 2016 Jolla Ltd.
```

**Comment by paveltkv** [comment]
Added Notice when geolocation is disabled.

File: import/popups/PopupOpener.qml [hunk 4]
```diff
@@ -34,6 +35,13 @@ Timer {
 
     property Component _contextMenuComponent
 
+    property Notice geolocationDisabledNotice: Notice {
+        duration: 3000
+        //% "Geolocation is disabled"
+        text: qsTrId("sailfish_components_webview_popupopener-la-geolocation_disabled")
+        verticalOffset: -overlay.toolBar.toolsHeight * 4
```
Comment (rainemak) [inline import/popups/PopupOpener.qml:42]: This cannot be like this as this is declared in the browser. Email or any other embedder would fail.

One idea is that sailfish-components-webview has a qml singleton for theming - WebViewTheme like component (not sure what we'd call it). Browser could then override that notificationOffset value. Default could that -Theme.itemSizeMedium.

To be noted as well that TextSelectionController has Notice component.

File: import/popups/LocationDialog.qml [hunk 5]
```diff
@@ -17,17 +18,42 @@ UserPrompt {
...
+    //: Deny the server to use location
+    //% "Deny"
+    cancelText: qsTrId("sailfish_components_webview_popups-he-deny_location")
+
+    Column {
+        width: parent.width
+        spacing: Theme.paddingMedium
+
+        Label {
+            x: Theme.paddingLarge
```
Comment (rainemak) [inline import/popups/LocationDialog.qml:36]:
```suggestion
            x: Theme.horizontalPageMargin
```
This should be upgraded to Theme.horizontalPageMargin

File: import/popups/LocationDialog.qml [hunk 6]
```diff
@@ -17,17 +18,42 @@ UserPrompt {
...
+            width: parent.width - Theme.paddingLarge * 2
```
Comment (rainemak) [inline import/popups/LocationDialog.qml:37]:
```suggestion
            width: parent.width - x * 2
```

File: import/popups/PopupOpener.qml [hunk 7]
```diff
@@ -1,6 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2016 Jolla Ltd.
+** Copyright (c) 2016 Jolla Ltd.
```
Comment (rainemak) [inline import/popups/PopupOpener.qml:3]:
```suggestion
** Copyright (c) 2016 - 2017 Jolla Ltd.
```

File: import/popups/LocationDialog.qml [hunk 8]
```diff
@@ -17,17 +18,42 @@ UserPrompt {
...
+
+            wrapMode: Text.WordWrap
+            textFormat: Text.StyledText
+            color: Theme.highlightColor
+        }
+
+        TextSwitch {
+            id: remember
+
+            checked: true
```
Comment (rainemak) [inline import/popups/LocationDialog.qml:52]: Let's do not default to true until we have a way to undo the change.

File: import/popups/LocationDialog.qml [hunk 9]
```diff
@@ -17,17 +18,42 @@ UserPrompt {
...
+        spacing: Theme.paddingMedium
+
+        Label {
+            x: Theme.horizontalPageMargin
+            width: parent.width - x * 2
+
+            //: %1 is the site that wants know user location
+            //% "Allow %1 to use your location?"
+            text: qsTrId("sailfish_components_webview_popups-la-location_request")
+                        .arg("<font color=\"" + Theme.rgba(Theme.primaryColor, Theme.opacityHigh) + "\">" + host + "</font>")
```
Comment (rainemak) [inline import/popups/LocationDialog.qml:42]: Please remove the underlining so that this is not mixed with actual links.

Just highlightColored text implying that special formatting is not needed either.

File: import/popups/LocationDialog.qml [hunk 10]
```diff
@@ -17,17 +18,42 @@ UserPrompt {
...
+
+            wrapMode: Text.WordWrap
+            textFormat: Text.StyledText
```
Comment (rainemak) [inline import/popups/LocationDialog.qml:45]: Drop this

File: import/popups/LocationDialog.qml (hunk 9 above)
Comment (rainemak) [reply import/popups/LocationDialog.qml:42]: Meant that let's have the whole text with highlightColor.

File: import/popups/LocationDialog.qml [hunk 11]
```diff
@@ -17,17 +18,41 @@ UserPrompt {
...
+                        .arg(host)
+
+            wrapMode: Text.WordWrap
+            color: Theme.highlightColor
+        }
+
+        TextSwitch {
+            id: remember
+
+            checked: false
```
Comment (rainemak) [inline import/popups/LocationDialog.qml:51]: Default value is false

**Review by rainemak — CHANGES_REQUESTED** [review]
Minor note but after that this is good to go.

**Review by rainemak — APPROVED** [review]
Thanks for the prompt update. Approved.
